Update README with migrations info (#353)
equinor/flotilla: README.md

git clone https://github.com/equinor/flotilla
```

Please see separate installation guides for [frontend](frontend), [backend](backend), and [Broker](broker).
For the environment setup, either run the script as described below or do it manually as described in each component.

### Automatic environment setup

Run the [setup.sh](./setup.sh) to automatically set up your dev environment for the components.
This script will ask you for the `Client Secret` for the backend and the `MQTT broker server key` for the MQTT broker.

## Database model and EF Core

Our database model is defined in the folder
[`/backend/api/Database/Models`](/backend/api/Database/Models) and we use

> [Entity FrameWork Core](https://docs.microsoft.com/en-us/ef/core/) as an

Reviewer: ```suggestion
[Entity Framework Core](https://docs.microsoft.com/en-us/ef/core/) as an
```

Author: Fixed

object-relational mapper (O/RM). When making changes to the model, we also need
to create a new
[migration](https://docs.microsoft.com/en-us/ef/core/managing-schemas/migrations/)
and apply it to our databases.

### Installing EF Core

```bash
dotnet tool install --global dotnet-ef
```

### Creating a new migration

> After making changes to the model run (from `/backend/api`):

Reviewer: ```suggestion
After making changes to the model, run the following command from `/backend/api`:
```

Author: Fixed

```bash
dotnet ef migrations add {migration-name}
```

`add` will make changes to existing files and add 2 new files in
`backend/api/Migrations`, which all need to be checked in to git.

Note that the {migration-name} is just a descriptive name of your choosing.
Also note that `Database__ConnectionString` should be pointed at one of our
databases when running `add`. The reason for this is that the migration will be
created slightly different when based of the in-memory database. `add` will _not_
update or alter the connected database in any way.

If you for some reason are unhappy with your migration, you can delete it with

```bash
dotnet ef migrations remove
```

> or simply delete the files and changes created by `add`. Once deleted you can

Reviewer: I don't think this is recommended or as safe as running the `remove` command, maybe we shouldn't encourage it.

Author: Agreed!

git clone https://github.com/equinor/flotilla
```

Please see separate installation guides for [frontend](frontend), [backend](backend), and [Broker](broker).
For the environment setup, either run the script as described below or do it manually as described in each component.

### Automatic environment setup

Run the [setup.sh](./setup.sh) to automatically set up your dev environment for the components.
This script will ask you for the `Client Secret` for the backend and the `MQTT broker server key` for the MQTT broker.

## Database model and EF Core

Our database model is defined in the folder
[`/backend/api/Database/Models`](/backend/api/Database/Models) and we use
[Entity Framework Core](https://docs.microsoft.com/en-us/ef/core/) as an
object-relational mapper (O/RM). When making changes to the model, we also need
to create a new
[migration](https://docs.microsoft.com/en-us/ef/core/managing-schemas/migrations/)
and apply it to our databases.

### Installing EF Core

```bash
dotnet tool install --global dotnet-ef
```

### Creating a new migration

> Make sure you have have fetched the newest code from main and that noone else
> is making migrations at the same time as you.

Reviewer: ```suggestion

**NB: Make sure you have have fetched the newest code from main and that no-one else
is making migrations at the same time as you!**  

```

Author: Fixed!
